File: rag-powered-inventory-management/lovable-page/src/data/inventory.ts
export interface InventoryItem {
  sku: string;
  name: string;
  category: string;
  productType: string;
  active: boolean;
  stock: number;
  averageDailyDemand: number;
  coverageDays: number | null;
  coverageLabel: string;
  stockStatus: string;
  lastUpdated: string;
  sourceText: string;
}

export interface InventorySummary {
  asOf: string;
  totalProducts: number;
  activeProducts: number;
  inactiveProducts: number;
  totalUnits: number;
  criticalProducts: number;
  lowProducts: number;
  healthyProducts: number;
}

export interface InventoryResponse {
  summary: InventorySummary;
  items: InventoryItem[];
}

export interface ChatResponse {
  answer: string;
}

export function formatCoverage(item: InventoryItem) {
  if (item.coverageDays === null) return item.coverageLabel;
  return `${item.coverageDays.toFixed(2)} days`;
}

File: backend/schemas/inventory.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class InventoryRecommendation(BaseModel):
    sku: str
    name: str
    current_stock: int = 0
    min_stock: float = 0
    max_stock: float = 0
    reorder_point: float = 0
    recommended_order: int = 0
    order_urgency: str = "normal"
    expected_lead_time_days: float = 0
    notes: str = ""


class InventoryItemOut(BaseModel):
    sku: str
    name: str
    category: str
    productType: str = ""
    active: bool = True
    stock: int = 0
    averageDailyDemand: float = 0.0
    coverageDays: float | None = None
    coverageLabel: str = ""
    stockStatus: str = "Healthy"
    lastUpdated: str = ""
    sourceText: str = ""


class InventorySummaryOut(BaseModel):
    asOf: str = ""
    totalProducts: int = 0
    activeProducts: int = 0
    inactiveProducts: int = 0
    totalUnits: int = 0
    criticalProducts: int = 0
    lowProducts: int = 0
    healthyProducts: int = 0
